packages/hawtio/src/plugins/camel/CamelPreferences.tsx

```diff
@@ -0,0 +1,111 @@
+import { CardBody, Checkbox, Form, FormGroup, FormSection, TextInput } from '@patternfly/react-core'
+import React, { useState } from 'react'
+import { camelPreferencesService, ICamelPreferences } from './camel-preferences-service'
+
+export const CamelPreferences: React.FunctionComponent = () => (
+  <CardBody>
+    <Form isHorizontal>
+      <CamelPreferencesForm />
+    </Form>
+  </CardBody>
+)
+
+const CamelPreferencesForm: React.FunctionComponent = () => {
+  const [camelPreferences, setCamelPreferences] = useState(camelPreferencesService.loadCamelPreferences())
+
+  const updatePreferences = (value: boolean | number, key: keyof ICamelPreferences): void => {
+    const updatedPreferences = { ...camelPreferences, ...{ [key]: value } }
+
+    camelPreferencesService.saveCamelPreferences(updatedPreferences)
+    setCamelPreferences(updatedPreferences)
+  }
+
+  const updateNumberValueFor = (key: keyof ICamelPreferences): ((value: string) => void) => {
```

Comment:
This might seem unorthodox but allows to not have to put arrow functions below and just hide this bit of configuration code. I don't know if it might be too complex of a concept, I'd like your input on this.